internal/model/function_test.go

```diff
@@ -0,0 +1,221 @@
+package model_test
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+
+	"github.com/go-logr/logr"
+
+	"github.com/theunrepentantgeek/crddoc/internal/config"
+	"github.com/theunrepentantgeek/crddoc/internal/model"
+	"github.com/theunrepentantgeek/crddoc/internal/packageloader"
+)
+
+func TestObject_Functions_ReturnsExpectedContent(t *testing.T) {
+	t.Parallel()
+	g := NewGomegaWithT(t)
+
+	cfg := &config.Config{}
+	loader := packageloader.New(cfg, logr.Discard())
+
+	pkg, err := loader.LoadFile(testdataPath(t, "person_with_methods.go"))
+	g.Expect(err).To(Succeed())
+	g.Expect(pkg).NotTo(BeNil())
+
+	dec, ok := pkg.Declaration("PersonWithMethods")
+	g.Expect(ok).To(BeTrue())
+
+	obj, ok := dec.(*model.Object)
+	g.Expect(ok).To(BeTrue())
+
+	// Verify we have the expected functions
+	functions := obj.Functions()
+	g.Expect(functions).To(HaveLen(5))
+
+	// Check function names
+	functionNames := make([]string, len(functions))
+	for i, fn := range functions {
+		functionNames[i] = fn.Name
+	}
+	g.Expect(functionNames).To(ConsistOf("GetName", "SetName", "IsAdult", "UpdateAge", "Compare"))
+}
+
+func TestObject_Function_ReturnsExpectedFunction(t *testing.T) {
+	t.Parallel()
+	g := NewGomegaWithT(t)
+
+	cases := map[string]struct {
+		functionName   string
+		expectedExists bool
+		checkDetails   func(*testing.T, *model.Function)
```

Comment:
Change this to take the gomega `g` so that all the assertions in the subtest are handled by the same gomega instance.